packages/frontend/src/components/common/modal/DeleteActionModal.tsx

```diff
@@ -1,55 +1,103 @@
-import React, { Dispatch, SetStateAction, useEffect } from 'react';
-import { UseMutationResult } from 'react-query';
-import ActionModal, { ActionModalProps } from './ActionModal';
+import { Button, Classes, Dialog } from '@blueprintjs/core';
+import { DashboardBasicDetails } from 'common';
+import { FC, useEffect, useState } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import { useDeleteMutation as useDeleteDashboardMutation } from '../../../hooks/dashboard/useDashboard';
+import { getDashboards } from '../../../hooks/dashboard/useDashboards';
+import { useDeleteMutation } from '../../../hooks/useSavedQuery';
 
-type DeleteActionModalProps<T> = {
-    useActionModalState: [
-        { actionType: number; data?: T },
-        Dispatch<SetStateAction<{ actionType: number; data?: T }>>,
-    ];
-    useDelete: UseMutationResult<undefined, ApiError, string>;
-    ModalContent: (
-        props: Pick<ActionModalProps<T>, 'useActionModalState' | 'isDisabled'>,
-    ) => JSX.Element;
-};
+interface DeleteActionModalProps {
+    name: string;
+    uuid: string;
+    isOpen: boolean;
+    isChart: boolean;
+    onClose: () => void;
+}
 
-const DeleteActionModal = <T extends { uuid: string; name: string }>(
-    props: DeleteActionModalProps<T>,
-) => {
-    const { useDelete, useActionModalState, ModalContent } = props;
-    const [actionState] = useActionModalState;
-    const { data: { uuid: id } = {}, actionType } = actionState;
-    const {
-        status: statusDelete,
-        mutate: deleteData,
-        isLoading: isDeleting,
-        reset: resetDelete,
-    } = useDelete;
+const DeleteActionModal: FC<DeleteActionModalProps> = ({
+    name,
+    uuid,
+    isOpen,
+    onClose,
+    isChart,
+}) => {
+    const { projectUuid } = useParams<{ projectUuid: string }>();
+    const { mutate: deleteDashboard, isLoading: isDeleting } =
+        useDeleteDashboardMutation();
+
+    const { mutate: deleteChart, isLoading } = useDeleteMutation();
+    const [relatedDashboards, setRelatedDashboards] = useState<
+        DashboardBasicDetails[]
+    >([]);
 
-    const onSubmitForm = () => {
-        if (id) {
-            deleteData(id);
-        }
-    };
     useEffect(() => {
-        if (!isDeleting) {
-            resetDelete();
+        if (isChart) {
+            getDashboards(projectUuid, uuid).then((dashboards) => {
+                setRelatedDashboards(dashboards);
+            });
         }
-    }, [isDeleting, actionType, resetDelete]);
+    }, [isChart, uuid, projectUuid]);
 
     return (
-        <ActionModal
-            title="Delete"
-            confirmButtonLabel="Delete"
-            confirmButtonIntent={Intent.DANGER}
-            useActionModalState={useActionModalState}
-            isDisabled={isDeleting}
-            onSubmitForm={onSubmitForm}
-            completedMutation={statusDelete === 'success'}
-            ModalContent={ModalContent}
-        />
+        <Dialog
```

Comment:
We should refactor & move the dialog from `packages/frontend/src/components/Explorer/VisualizationCard/VisualizationCard.tsx` into it's own component so we can reuse it here instead of duplicating all the code.